File: src/main/java/com/optica/services/CompanyServiceImpl.java
package com.optica.services;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.optica.domain.Company;
import com.optica.repository.CompanyDao;

@Service
public class CompanyServiceImpl implements CompanyService {

	@Autowired
	private CompanyDao companyDao;

	@Override
	public Company findCompany(Long id) {
		return companyDao.findById(id).get();
	}

	@Override
	public List<Company> findAllCompanies() {
		return companyDao.findAll();
	}

}

File: src/main/kotlin/com/optica/services/CompanyServiceImpl.kt
package com.optica.services

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

import com.optica.domain.Company
import com.optica.repository.CompanyDao

@Service
class CompanyServiceImpl : CompanyService {

    @Autowired
    private val companyDao: CompanyDao? = null

    override fun findCompany(id: Long?): Company {
        return companyDao!!.findById(id!!).get()
    }

    override fun findAllCompanies(): List<Company> {
        return companyDao!!.findAll()
    }

}
